File: backend/app/api/tasks.py
from flask import jsonify, request
from bson import ObjectId
from app.models import Task
from app.api import bp

@bp.route('/tasks', methods=['GET'])
def get_tasks():
    tasks = Task.all()
    return jsonify([task.to_dict() for task in tasks])

@bp.route('/tasks', methods=['POST'])
def create_task():
    data = request.get_json() or {}
    if 'title' not in data:
        return jsonify({'error': 'Missing required field: title'}), 400
    
    task = Task(
        title=data['title'],
        description=data.get('description', '')
    )
    task.save()
    return jsonify(task.to_dict()), 201

@bp.route('/tasks/<task_id>', methods=['GET'])
def get_task(task_id):
    try:
        object_id = ObjectId(task_id)
    except:
        return jsonify({'error': 'Invalid task ID'}), 400
    
    task = Task.get_or_404(object_id)
    return jsonify(task.to_dict())

@bp.route('/tasks/<task_id>', methods=['PUT'])
def update_task(task_id):
    try:
        object_id = ObjectId(task_id)
    except:
        return jsonify({'error': 'Invalid task ID'}), 400
    
    task = Task.get_or_404(object_id)
    data = request.get_json() or {}
    
    if 'title' in data:
        task.title = data['title']
    if 'description' in data:
        task.description = data['description']
    
    task.save()
    return jsonify(task.to_dict())

@bp.route('/tasks/<task_id>', methods=['DELETE'])
def delete_task(task_id):
    try:
        object_id = ObjectId(task_id)
    except:
        return jsonify({'error': 'Invalid task ID'}), 400
        
    task = Task.get_or_404(object_id)
    task.delete()
    return '', 204

File: flask-react-assessment-ashishagarwal10-main/src/apps/frontend/api/tasks.ts
import { apiClient } from './client';

export interface Task {
    id: string;
    title: string;
    description: string;
    completed: boolean;
    created_at: string;
    updated_at: string;
}

export interface TaskCreateData {
    title: string;
    description: string;
}

export const getTasks = async () => {
    return await apiClient.get<Task[]>('/tasks');
};

export const getTask = async (id: string) => {
    return await apiClient.get<Task>(`/tasks/${id}`);
};

export const createTask = async (data: TaskCreateData) => {
    return await apiClient.post<Task>('/tasks', data);
};

export const updateTask = async (id: string, data: Partial<TaskCreateData> & { completed?: boolean }) => {
    return await apiClient.put<Task>(`/tasks/${id}`, data);
};

export const deleteTask = async (id: string) => {
    return await apiClient.delete(`/tasks/${id}`);
};